src/main/java/com/github/wvengen/maven/proguard/ProGuardMojo.java

```diff
@@ -693,6 +707,32 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 			Collections.addAll(args, options);
 		}
 
+
+		if(generateTemporaryConfigurationFile) {
+			log.info("building config file");
+
+			StringBuilder stringBuilder = new StringBuilder();
+			for (String arg : args) {
+				if (arg.startsWith("-")) {
+					stringBuilder.append("\n");
+				}else{
+					stringBuilder.append(" ");
+				}
+				stringBuilder.append(arg);
+			}
+
+			try {
+				FileWriter writer = new FileWriter(temporaryConfigurationFile);
+				IOUtils.write(stringBuilder.toString(), writer);
+				writer.close();
```

Comment:
Writer needs closing even in case of errors, so this needs to be done in a finally block or with 
```
try(FileWriter write = new FileWriter(temporaryConfigurationFile); ) {

}
```

Reply:
I replaced it with a java 1.6 compatible finally.